resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -736,22 +734,20 @@ private boolean doResolveAllCached(String hostname,
 
         List<InetAddress> result = null;
         Throwable cause = null;
-        synchronized (cachedEntries) {
-            final int numEntries = cachedEntries.size();
-            assert numEntries > 0;
+        final int numEntries = cachedEntries.size();
```

Comment:
move this into the conditional block where it is actually used. If the assertion is not there `get(0)` will throw, and we already check for `isEmpty()` above.